agent/tcs/client/client_test.go

```diff
@@ -46,6 +46,8 @@ const (
 	rwTimeout                  = time.Second
 )
 
+var testCreds = credentials.NewStaticCredentials("test-id", "test-secret", "test-token")
```

Comment:
Should we add a comment here like "Non-empty credentials should be used to avoid signing failure"? Otherwise it's not obvious that AnonymousCredentials can't be used in these tests..

Reply:
actually maybe ill just add it to the PR description. cause we cant add that comment to every test and if you try to use it in this context, the tests will fail for these code paths.